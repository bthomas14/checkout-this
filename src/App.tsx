/** @jsxRuntime classic */
/** @jsx jsx */
import {jsx} from '@emotion/react';
import axios from 'axios';
import React, {useEffect, useState} from 'react';

import Comment from './Components/Comment';
import ReviewChart from './Components/ReviewChart';
import ReviewForm from './Components/ReviewForm';
import {Review} from './types';

/*
  THEME
  primary:    #2bd4db
  border:     #cecfd9
  text:       #0c1142
*/

// In a real world app, we'd be getting this id from the url
const PRODUCT_ID = 1;

const App: React.FunctionComponent = () => {
  const [reviews, setReviews] = useState<Review[]>([]);
  const [serverError, setServerError] = useState(false);

  const fetchReviews = async () => {
    setServerError(false);
    try {
      const res = await axios.get(`/reviews?product_id=${PRODUCT_ID}&_sort=createdAt&_order=desc&_limit=20`);
      setReviews(res.data);
    } catch (err) {
      console.error('fetchReviews.Error', err);
      setServerError(true);
    }
  };

  useEffect(() => {
    void fetchReviews();
  }, []);

  return (
    <div css={{margin: '40px'}}>
      <h1 css={{fontWeight: '600'}}>Write a review</h1>
      <div 
        css={{
          display: 'flex', 
          justifyContent: 'space-between',
          '@media (max-width: 1024px)': {
            flexDirection: 'column',
          },
        }}
      >
        <div 
          css={{
            width: '40%',
            '@media (max-width: 1024px)': {
              width: '100%',
            },
          }}>
          <ReviewForm productId={PRODUCT_ID} onSubmit={fetchReviews} />
        </div>

        <div 
          css={{
            position: 'relative', 
            width: '600px', 
            height: '300px',
            '@media (max-width: 1024px)': {
              marginTop: '30px',
            },
          }}
        >
          <ReviewChart reviews={reviews} />
        </div>
      </div>

      <div css={{paddingTop: '40px'}}>
        <h2 css={{fontWeight: '600'}}>Latest Comments</h2>
        {serverError ? (
          <div>
            Oops, something went wrong. Try <button onClick={fetchReviews}>refreshing</button>.
          </div>
        ) : (
          <React.Fragment>
            {reviews.length > 0 && reviews.map((review) => <Comment key={review.id} review={review} />)}
          </React.Fragment>
        )}
      </div>
    </div>
  );
};

export default App;
