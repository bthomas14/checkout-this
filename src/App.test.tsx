import React from 'react';
import {render} from '@testing-library/react';

import App from './App';

test('Renders app successfully', () => {
  const app = render(<App />);
  expect(app).toMatchSnapshot();
});
