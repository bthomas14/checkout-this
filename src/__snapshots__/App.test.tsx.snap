// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders app successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1cxukqa-App"
      >
        <h1
          class="css-17femtr-App"
        >
          Write a review
        </h1>
        <div
          class="css-pto0l7-App"
        >
          <div
            class="css-1ooilkz-App"
          >
            <form
              data-testid="review-form"
            >
              <div>
                <div
                  class="css-upqvjc-Input"
                >
                  <label
                    class="css-1yrktxg-Input"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    aria-label="Name"
                    class="css-x5ipqg-Input"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="css-upqvjc-Input"
              >
                <label
                  class="css-1yrktxg-Input"
                  for="email"
                >
                  Email
                </label>
                <input
                  aria-label="Email"
                  class="css-x5ipqg-Input"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="css-1mphgo4-StarRating"
              >
                <label
                  class="css-vnkehg-StarRating"
                  for="rating"
                >
                  Rating
                </label>
                <ul
                  class="css-9jn6y5-StarRating"
                >
                  <li
                    class="css-1pzx6l1-StarRating"
                    data-testid="rating-1"
                  >
                    <div
                      class="unselected css-aip1w-StarRating"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star "
                        color="#fff"
                        data-icon="star"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    class="css-1pzx6l1-StarRating"
                    data-testid="rating-2"
                  >
                    <div
                      class="unselected css-aip1w-StarRating"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star "
                        color="#fff"
                        data-icon="star"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    class="css-1pzx6l1-StarRating"
                    data-testid="rating-3"
                  >
                    <div
                      class="unselected css-aip1w-StarRating"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star "
                        color="#fff"
                        data-icon="star"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    class="css-1pzx6l1-StarRating"
                    data-testid="rating-4"
                  >
                    <div
                      class="unselected css-aip1w-StarRating"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star "
                        color="#fff"
                        data-icon="star"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    class="css-1pzx6l1-StarRating"
                    data-testid="rating-5"
                  >
                    <div
                      class="unselected css-aip1w-StarRating"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star "
                        color="#fff"
                        data-icon="star"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </li>
                </ul>
              </div>
              <div
                class="css-op039d-ReviewForm"
              >
                <label
                  class="css-166ulrh-ReviewForm"
                  for="comment"
                >
                  Comment
                </label>
                <textarea
                  aria-label="Comment"
                  class="css-1o6nvus-ReviewForm"
                  name="comment"
                  placeholder="Tell us what you thought..."
                  rows="5"
                  spellcheck="false"
                />
              </div>
              <button
                class="css-ala3ha-ReviewForm"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
          <div
            class="css-bzd9xv-App"
          >
            <div
              class="css-xteqtn-ReviewChart"
            >
              <div
                class="css-c0ufoy-ReviewChart"
              >
                <div
                  class="css-3qivuf-ReviewChart"
                  data-testid="avg-rating"
                >
                  NaN
                   stars out of 5
                </div>
                <div
                  data-testid="rating-count"
                >
                  0
                   reviews
                </div>
              </div>
              <div
                class="recharts-wrapper"
                style="position: relative; cursor: default; width: 400px; height: 200px;"
              >
                <svg
                  class="recharts-surface"
                  height="200"
                  version="1.1"
                  viewBox="0 0 400 200"
                  width="400"
                >
                  <defs>
                    <clippath
                      id="recharts1-clip"
                    >
                      <rect
                        height="180"
                        width="320"
                        x="70"
                        y="10"
                      />
                    </clippath>
                  </defs>
                  <g
                    class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
                  >
                    <g
                      class="recharts-cartesian-axis-ticks"
                    >
                      <g
                        class="recharts-layer recharts-cartesian-axis-tick"
                      >
                        <text
                          class="recharts-text recharts-cartesian-axis-tick-value"
                          fill="#666"
                          height="180"
                          orientation="left"
                          stroke="none"
                          text-anchor="end"
                          type="number"
                          width="60"
                          x="62"
                          y="190"
                        >
                          <tspan
                            dy="0.355em"
                            x="62"
                          >
                            1 star
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis-tick"
                      >
                        <text
                          class="recharts-text recharts-cartesian-axis-tick-value"
                          fill="#666"
                          height="180"
                          orientation="left"
                          stroke="none"
                          text-anchor="end"
                          type="number"
                          width="60"
                          x="62"
                          y="145"
                        >
                          <tspan
                            dy="0.355em"
                            x="62"
                          >
                            2 star
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis-tick"
                      >
                        <text
                          class="recharts-text recharts-cartesian-axis-tick-value"
                          fill="#666"
                          height="180"
                          orientation="left"
                          stroke="none"
                          text-anchor="end"
                          type="number"
                          width="60"
                          x="62"
                          y="100"
                        >
                          <tspan
                            dy="0.355em"
                            x="62"
                          >
                            3 star
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis-tick"
                      >
                        <text
                          class="recharts-text recharts-cartesian-axis-tick-value"
                          fill="#666"
                          height="180"
                          orientation="left"
                          stroke="none"
                          text-anchor="end"
                          type="number"
                          width="60"
                          x="62"
                          y="55"
                        >
                          <tspan
                            dy="0.355em"
                            x="62"
                          >
                            4 star
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis-tick"
                      >
                        <text
                          class="recharts-text recharts-cartesian-axis-tick-value"
                          fill="#666"
                          height="180"
                          orientation="left"
                          stroke="none"
                          text-anchor="end"
                          type="number"
                          width="60"
                          x="62"
                          y="10"
                        >
                          <tspan
                            dy="0.355em"
                            x="62"
                          >
                            5 star
                          </tspan>
                        </text>
                      </g>
                    </g>
                  </g>
                  <g
                    class="recharts-layer recharts-bar"
                  >
                    <g
                      class="recharts-layer recharts-bar-rectangles"
                    >
                      <g
                        class="recharts-layer"
                      >
                        <g
                          class="recharts-layer recharts-bar-rectangle"
                        />
                        <g
                          class="recharts-layer recharts-bar-rectangle"
                        />
                        <g
                          class="recharts-layer recharts-bar-rectangle"
                        />
                        <g
                          class="recharts-layer recharts-bar-rectangle"
                        />
                        <g
                          class="recharts-layer recharts-bar-rectangle"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
                <div
                  class="recharts-tooltip-wrapper"
                  style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-14fdefm-App"
        >
          <h2
            class="css-17femtr-App"
          >
            Latest Comments
          </h2>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      id="recharts_measurement_span"
      style="position: absolute; top: -20000px; left: 0px; padding: 0px; margin: 0px; white-space: pre;"
    >
      5
    </span>
  </body>,
  "container": <div>
    <div
      class="css-1cxukqa-App"
    >
      <h1
        class="css-17femtr-App"
      >
        Write a review
      </h1>
      <div
        class="css-pto0l7-App"
      >
        <div
          class="css-1ooilkz-App"
        >
          <form
            data-testid="review-form"
          >
            <div>
              <div
                class="css-upqvjc-Input"
              >
                <label
                  class="css-1yrktxg-Input"
                  for="name"
                >
                  Name
                </label>
                <input
                  aria-label="Name"
                  class="css-x5ipqg-Input"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="css-upqvjc-Input"
            >
              <label
                class="css-1yrktxg-Input"
                for="email"
              >
                Email
              </label>
              <input
                aria-label="Email"
                class="css-x5ipqg-Input"
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
            <div
              class="css-1mphgo4-StarRating"
            >
              <label
                class="css-vnkehg-StarRating"
                for="rating"
              >
                Rating
              </label>
              <ul
                class="css-9jn6y5-StarRating"
              >
                <li
                  class="css-1pzx6l1-StarRating"
                  data-testid="rating-1"
                >
                  <div
                    class="unselected css-aip1w-StarRating"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star "
                      color="#fff"
                      data-icon="star"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  class="css-1pzx6l1-StarRating"
                  data-testid="rating-2"
                >
                  <div
                    class="unselected css-aip1w-StarRating"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star "
                      color="#fff"
                      data-icon="star"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  class="css-1pzx6l1-StarRating"
                  data-testid="rating-3"
                >
                  <div
                    class="unselected css-aip1w-StarRating"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star "
                      color="#fff"
                      data-icon="star"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  class="css-1pzx6l1-StarRating"
                  data-testid="rating-4"
                >
                  <div
                    class="unselected css-aip1w-StarRating"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star "
                      color="#fff"
                      data-icon="star"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  class="css-1pzx6l1-StarRating"
                  data-testid="rating-5"
                >
                  <div
                    class="unselected css-aip1w-StarRating"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star "
                      color="#fff"
                      data-icon="star"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="css-op039d-ReviewForm"
            >
              <label
                class="css-166ulrh-ReviewForm"
                for="comment"
              >
                Comment
              </label>
              <textarea
                aria-label="Comment"
                class="css-1o6nvus-ReviewForm"
                name="comment"
                placeholder="Tell us what you thought..."
                rows="5"
                spellcheck="false"
              />
            </div>
            <button
              class="css-ala3ha-ReviewForm"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          class="css-bzd9xv-App"
        >
          <div
            class="css-xteqtn-ReviewChart"
          >
            <div
              class="css-c0ufoy-ReviewChart"
            >
              <div
                class="css-3qivuf-ReviewChart"
                data-testid="avg-rating"
              >
                NaN
                 stars out of 5
              </div>
              <div
                data-testid="rating-count"
              >
                0
                 reviews
              </div>
            </div>
            <div
              class="recharts-wrapper"
              style="position: relative; cursor: default; width: 400px; height: 200px;"
            >
              <svg
                class="recharts-surface"
                height="200"
                version="1.1"
                viewBox="0 0 400 200"
                width="400"
              >
                <defs>
                  <clippath
                    id="recharts1-clip"
                  >
                    <rect
                      height="180"
                      width="320"
                      x="70"
                      y="10"
                    />
                  </clippath>
                </defs>
                <g
                  class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
                >
                  <g
                    class="recharts-cartesian-axis-ticks"
                  >
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="180"
                        orientation="left"
                        stroke="none"
                        text-anchor="end"
                        type="number"
                        width="60"
                        x="62"
                        y="190"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          1 star
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="180"
                        orientation="left"
                        stroke="none"
                        text-anchor="end"
                        type="number"
                        width="60"
                        x="62"
                        y="145"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          2 star
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="180"
                        orientation="left"
                        stroke="none"
                        text-anchor="end"
                        type="number"
                        width="60"
                        x="62"
                        y="100"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          3 star
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="180"
                        orientation="left"
                        stroke="none"
                        text-anchor="end"
                        type="number"
                        width="60"
                        x="62"
                        y="55"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          4 star
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="180"
                        orientation="left"
                        stroke="none"
                        text-anchor="end"
                        type="number"
                        width="60"
                        x="62"
                        y="10"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          5 star
                        </tspan>
                      </text>
                    </g>
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-bar"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangles"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                    </g>
                  </g>
                </g>
              </svg>
              <div
                class="recharts-tooltip-wrapper"
                style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-14fdefm-App"
      >
        <h2
          class="css-17femtr-App"
        >
          Latest Comments
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
